### ISSUE TITLE
Don't use synchronous requests

### ISSUE THREAD

**Issue by brianmichel** [cedexis/RadarKit#4]
I've noticed in the `RadarKit` library, the common pattern is to do something like this...

``` objc
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
        [[[Radar alloc] init]
            runForZoneId:1
           AndCustomerId:10660
             //AndProtocol:@"https"
        ];
    });
```

From my reading, it seems like this is bacause `+ (NSData *)sendSynchronousRequest:(NSURLRequest *)request returningResponse:(NSURLResponse **)response error:(NSError **)error` is used internally to the library to perform network requests.

From the calling interface, i.e. `[[[Radar alloc] init] ...]` there is no return value, and not asynchronous callback to pass in, so it's a bit of a smell to run this in it's own dispatch block.

Would it be possible to move to using `+ (void)sendAsynchronousRequest:(NSURLRequest *)request queue:(NSOperationQueue *)queue completionHandler:(void (^)(NSURLResponse *response, NSData *data, NSError *connectionError))handler` internally? This would allow the caller to not dispatch into a background block while calling into your library.

**Comment by jakewan** [comment]
"This would allow the caller to not dispatch into a background block while calling into your library."

That's a great argument. Will do.

**Comment by brianmichel** [comment]
:+1:

**Comment by javorosas** [comment]
Resolved in 20602f55a115ee7f3bce59de7f369e22f70805eb
